contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/JsonTableGroupScan.java

```diff
@@ -100,30 +104,46 @@ public GroupScan clone(List<SchemaPath> columns) {
     return newScan;
   }
 
+  public JsonTableGroupScan clone(JsonScanSpec scanSpec) {
+    JsonTableGroupScan newScan = new JsonTableGroupScan(this);
+    newScan.scanSpec = scanSpec;
+    newScan.computeRegionsToScan();
+    return newScan;
+  }
+
+  private void computeRegionsToScan() {
+    boolean foundStartRegion = false;
+
+    regionsToScan = new TreeMap<TabletFragmentInfo, String>();
+    for (TabletInfo tabletInfo : tabletInfos) {
+      TabletInfoImpl tabletInfoImpl = (TabletInfoImpl) tabletInfo;
+      if (!foundStartRegion && !isNullOrEmpty(scanSpec.getStartRow()) && !tabletInfoImpl.containsRow(scanSpec.getStartRow())) {
+        continue;
+      }
+      foundStartRegion = true;
+      regionsToScan.put(new TabletFragmentInfo(tabletInfoImpl), tabletInfo.getLocations()[0]);
+      if (!isNullOrEmpty(scanSpec.getStopRow()) && tabletInfoImpl.containsRow(scanSpec.getStopRow())) {
+        break;
+      }
+    }
+  }
+
   private void init() {
     logger.debug("Getting tablet locations");
     try {
       Configuration conf = new Configuration();
-      Table t = MapRDB.getTable(scanSpec.getTableName());
-      TabletInfo[] tabletInfos = t.getTabletInfos(scanSpec.getCondition());
-      tableStats = new MapRDBTableStats(conf, scanSpec.getTableName());
 
-      boolean foundStartRegion = false;
-      regionsToScan = new TreeMap<TabletFragmentInfo, String>();
+      // Fetch table and tabletInfo only once and cache.
+      table = MapRDB.getTable(scanSpec.getTableName());
+      tabletInfos = table.getTabletInfos(scanSpec.getCondition());
+
+      // Calculate totalRowCount for the table
```

Comment:
Please add a comment explaining why we compute the totalRowCount like so? 
`totalRowCount += tabletInfo.getEstimatedNumRows();`